### PR TITLE
server CA persistence support

### PR DESCRIPTION
**Pull request by azdagron**
- updates memory CA to optionally persist signing keypair to disk.
- updates CA manager to skip the initial rotation if there is a good
  keypair returned by the CA.

Signed-off-by: Andrew Harding <azdagron@gmail.com>

### CODE REVIEW

File: doc/plugin_server_ca_memory.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
 # Server plugin: ServerCA "memory"
```
Comment (evan2645) [inline doc/plugin_server_ca_memory.md]: Is the name `memory` still accurate if the plugin can also persist to disk?

For now I think this is fine, but after upcoming ServerCA refactors we may want to consider making the disk-based functionality a dedicated plugin

File: doc/plugin_server_ca_memory.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 
-The `memory` plugin implements an in-memory signing authority. No keys are persisted to disk, and
-if the server is restarted, a new signing authority is generated against the upstream CA.
+The `memory` plugin implements an in-memory signing authority. The signing
+keypair is optionally persisted to disk. When the server is loaded, if no
+keypair has been persisted, or if the keypair is expiring/expired, a new
```
Comment (walmav) [inline doc/plugin_server_ca_memory.md]: how do we determine expiring?

Comment (azdagron) [reply doc/plugin_server_ca_memory.md]: Same way it is done today. Certificate is "expiring" when it is within 1/6th of the ttl left.

**Comment by walmav** [comment]
⚡️

File: doc/plugin_server_ca_memory.md (hunk 1 above)
Comment (azdagron) [reply doc/plugin_server_ca_memory.md]: Yep, i agree.

**Pull request merged**